kivy/tests/test_logger.py

```diff
@@ -0,0 +1,25 @@
+"""
+Logger tests
+============
+"""
+
+import unittest
+import pathlib
+
+from kivy.config import Config
+from kivy.logger import FileHandler
+
+
+class LoggerTest(unittest.TestCase):
+    def test_purge_logs(self):
+        maxfiles = 1
+        log_path = pathlib.Path("test_logs")
+
+        Config.set("kivy", "log_dir", log_path)
+        Config.set("kivy", "log_maxfiles", maxfiles)
+
+        FileHandler().purge_logs()
+
+        current_files = [f for f in log_path.iterdir() if f.is_file()]
+
+        self.assertEqual(maxfiles, len(current_files))
```

Comment:
```suggestion
import pytest
import pathlib
import time

@pytest.mark.parametrize('n', [0, 1, 5])
def test_purge_logs(tmp_path, n):
    from kivy.config import Config
    from kivy.logger import FileHandler

    Config.set("kivy", "log_dir", str(tmp_path))
    Config.set("kivy", "log_maxfiles", n)

    # create the default file first so it gets deleted so names match
    handler = FileHandler()
    handler._configure()
    # must close file so it can be deleted
    handler.fd.close()
    # wait a little so the timestamps are different for different files
    time.sleep(.001)

    names = [f'log_{i}.txt' for i in range(n + 2)]
    for name in names:
        p = tmp_path / name
        p.write_text('some data')
        time.sleep(.001)

    handler.purge_logs()

    # files that should have remained after purge
    expected_names = list(reversed(names))[:n]
    files = {f.name for f in tmp_path.iterdir()}
    assert expected_names == files

```

I re-wrote the test because we're using pytest when possible now. It has the benefit of it providing us with a temp directory for the test.

Also, we shouldn't create these files ahead of time, because it just adds space to the repo, instead, we can dynamically create them, like I do. So please remove the log files.